code/modules/vehicles/mecha/mech_melee_attack.dm

```diff
@@ -42,7 +42,14 @@
 			return 0
 	mecha_attacker.visible_message(span_danger("[mecha_attacker] hits [src]!"), span_danger("You hit [src]!"), null, COMBAT_MESSAGE_RANGE)
 	..()
-	return take_damage(mecha_attacker.force * 3, mecha_attacker.damtype, "melee", FALSE, get_dir(src, mecha_attacker)) // multiplied by 3 so we can hit objs hard but not be overpowered against mobs.
+	return take_damage(mecha_attacker.force * get_obj_mecha_damage_multiplier(), mecha_attacker.damtype, "melee", FALSE, get_dir(src, mecha_attacker))
+
+/** Gets the object mecha damage multiplier.
+ * Some objects (carp rift) shouldn't have a high multiplier, override this proc if necessary.
+ */
+/obj/proc/get_obj_mecha_damage_multiplier(obj/vehicle/sealed/mecha/mecha_attacker, mob/living/user)
```

Comment:
you added a proc to literally every object in the game for a multiplier on a single object.

no, use a signal with a return value or do not pass go.

Reply:
Signals can't return anything besides a bitflag

Reply:
Can have it return a bitflag with a value of 3 lol

Reply:
> Signals can't return anything besides a bitflag

bro you are tripped

Reply:
> Can have it return a bitflag with a value of 3 lol

you can physically return a bitflag but you are not intended to it goes against the design of signals. they are not meant to be used as proc calls

Reply:
You could pass the signal an empty list and allow consumers to add modifiers to the list

```dm
var/list/damage_mods = list(3)
SEND_SIGNAL(src, COMSIG_OBJ_MECHA_HIT, mecha_attacker, user, damage_mods)

var/final_damage = mecha_attacker.force
for(var/newmod in damage_mods)
    final_damage *= newmod
```

and

```
/obj/carp_rift/proc/on_mecha_hit(obj/item/source, obj/vechicle/mecha/attacker, mob/living/attacker_mob, list/damage_mods)
    SIGNAL_HANDLER
    damage_mods += 0.25 // 3 (base amount) * 0.25 = 0.27

```

Just a concept